tests/testthat/test-learn.R

```diff
@@ -1,74 +1,86 @@
 # FIXME: #313: learn only from current source
 
-test_that("Learning from MSSQL works?", {
+# produces a randomized schema name with a length of 4-10 characters
+# consisting of the symbols in `reservoir`
+random_schema <- function() {
+  reservoir <- c(letters, LETTERS, "'", "-", "_", as.character(0:9))
+  how_long <- sample(4:10, 1)
+  paste0(reservoir[sample(seq_len(length(reservoir)), how_long, replace = TRUE)], collapse = "")
+}
 
+schema_name <- random_schema()
 
+test_that("Standard learning from MSSQL (schema 'dbo') or Postgres (schema 'public') works?", {
+
+  skip_if_src_not(c("mssql", "postgres"))
   # dm_learn_from_mssql() --------------------------------------------------
-  src_mssql <- skip_if_error(src_test("mssql"))
+  src_db <- my_test_src()
 
   # create an object on the MSSQL-DB that can be learned
-  if (!any(src_tbls(src_mssql) %>%
-    grepl("^tf_1_", .))) {
-    copy_dm_to(src_mssql, dm_for_filter(), temporary = FALSE, table_names = unique_db_table_name)
-  }
-
-  dm_for_filter_mssql_learned <- dm_from_src(src_mssql)
-
-  def_learned_renamed_reclassed <-
-    dm_rename_tbl(
-      dm_for_filter_mssql_learned,
-      structure(src_tbls(dm_for_filter_mssql_learned), names = src_tbls(dm_for_filter()))
-    ) %>%
-    dm_get_def() %>%
-    select(-data)
-
-  def_original <-
-    dm_get_def(dm_for_filter()) %>%
-    select(-data)
-
-  expect_identical(
-    def_learned_renamed_reclassed,
-    def_original
+  dm_for_filter_copied <- copy_dm_to(src_db, dm_for_filter(), temporary = FALSE, table_names = ~ DBI::SQL(unique_db_table_name(.x)))
+  order_of_deletion <- c("tf_2", "tf_1", "tf_5", "tf_6", "tf_4", "tf_3")
+  remote_tbl_names <- map_chr(
+    order_of_deletion,
+    ~dbplyr::remote_name(dm_for_filter_copied[[.x]])
+  ) %>% set_names(order_of_deletion)
+
+  withr::defer(
+    walk(
+      dm_get_tables_impl(dm_for_filter_copied)[order_of_deletion],
+      ~ dbExecute(src_db$con, paste0("DROP TABLE ", dbplyr::remote_name(.x)))
+    )
+  )
+
+  dm_for_filter_db_learned_all <- dm_from_src(src_db)
+
+  # in case there happen to be other tables in schema "dbo" or "public"
+  dm_for_filter_db_learned <-
+    dm_for_filter_db_learned_all %>%
+    dm_select_tbl(!!!remote_tbl_names)
+
+  expect_equivalent_dm(
+    dm_for_filter_db_learned,
+    dm_for_filter()[order_of_deletion]
```

Comment:
Do we need to reorder?

```suggestion
    dm_for_filter()
```

Reply:
either reorder, or change the implementation of `expect_equivalent_dm()`; it fails currently, if the order of the tables in the `dm`-objects differ.